Mass-integration 0.20 to main

amirkl94 opened the pull request:
# PR title

Port release 0.20 branch fixes to `main`

tongyuantongyu committed:
fix: cleanup process tree for disaggregated test (#4116)

Signed-off-by: Yuan Tong <13075180+tongyuantongyu@users.noreply.github.com>

kaiyux committed:
chore: Remove unused script (#4485)

* Remove unused script

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

* Remove unused json file

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

---------

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

yuxianq committed:
chore: Deprecate autopp. (#4471)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

crazydemo committed:
tests: update api change from decoder to sampler in test (#4479)

update

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

Funatiq committed:
docs: Add KV Cache Management documentation (#3908)

* docs: Add KV Cache Management documentation

* Introduced a new document detailing the hierarchy and event system for KV cache management, including definitions for Pool, Block, and Page.
* Updated the index.rst to include a reference to the new kv-cache-management.md file.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* Update docs/source/advanced/kv-cache-management.md

Co-authored-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>
Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* Update KV Cache Pool Management

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Addcross-file links

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Clarify tokens_per_block

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

* docs: Clarify acronyms

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

---------

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>
Co-authored-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>

Tabrizian committed:
Add tritonrelease container (#4544)

Signed-off-by: Iman Tabrizian <10105175+tabrizian@users.noreply.github.com>

MartinMarciniszyn committed:
fix: [TRTLLM-325]WAR against security vulnerabilities in Python packages (#4539)

* fix: [TRTLLM-325]WAR against security vulnerabilities in Python packages

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

* Update docker images

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

---------

Signed-off-by: Martin Marciniszyn Mehringer <11665257+MartinMarciniszyn@users.noreply.github.com>

hyukn committed:
[5141290][5273694][5260696] fix: Fix mrope argument missing issue in the summary tasks for Qwen model. (#4432)

Fixed the mrope argument missing issue in the summary tasks for Qwen models.
And re-enabled the fixed tests.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

amirkl94 commented on .devcontainer/docker-compose.yml:
```diff
@@ -1,7 +1,7 @@
 services:
   tensorrt_llm-dev:
-    image: urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm:pytorch-25.04-py3-x86_64-ubuntu24.04-trt10.10.0.31-skip-tritondevel-202505191345-4400
+    image: urm.nvidia.com/sw-tensorrt-docker/tensorrt-llm:pytorch-25.04-py3-x86_64-ubuntu24.04-trt10.10.0.31-skip-tritondevel-202505211401-4539
```
@shaharmor98 I'm not sure about this image change but if you skim through to see I didn't miss anything

tensorrt-cicd commented:
[PR_Github #6153](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6153/) [ run ] triggered by Bot

amirkl94 commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6181](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6181/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #6181](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6181/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4520](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4520/) completed with status: 'FAILURE'

hyukn committed:
[5234029][5226211] chore: Unwaive multimodal tests for Qwen model. (#4519)

Unwaive multimodal tests for Qwen models.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

venkywonka committed:
test(perf): Extend the Llama-Nemotron-Nano-8B perf-integration-tests (pyt)  (#4407)

* extend pyt nano tests perf coverage

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

* explicitly set maxnt for some cases

This is because the test harness default to no prefill chunking, that means the isl specified is the true context.
When explicitly unspecified in the test harness, the `maxnt` passed down to `trtllm-bench` is 2048.
This means trtllm-bench gets conflicting inputs when isl>2048 but maxnt=2048; hence overriding maxnt to be consistent with isl for such cases.

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

---------

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

ruodil committed:
test: fix for perf sanity test and skip fp8 deepseek blackwell cases (#4598)

fix for sanity test and skip fp8 deepseek blackwell cases

Signed-off-by: Ruodi <200874449+ruodil@users.noreply.github.com>

hyukn committed:
[5180961] chore: Unwaive test for Qwen model. (#4524)

* Unwaive test for Qwen model.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

* update.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

---------

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

stnie committed:
ci: waive testcase [NVBUG 5297821] (#4616)

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

SimengLiu-nv committed:
[CI] Waive known errors with test TestDeepSeekV3Lite::test_fp8_block_scales_4gpus (#4627)

Signed-off-by: Simeng Liu <simengl@nvidia.com>

farazkh80 committed:
[TR[TLLM-4618][feat] Add remaining NVFP4 Nemotron Super 49B test on RTX6000 Pro (SM120) (#4548)

added nvfp4 nemotron for qa testing on RTX 6000

Signed-off-by: Faraz Khoubsirat <58580514+farazkh80@users.noreply.github.com>

moraxu committed:
[TRTLLM-4932] Add CLI accuracy tests for Llama-3_3-Nemotron-Super-49B-v1 and LLM API FP8 variant (#4375)

* Add CLI TestNemotronSuper acc tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update mmlu.yaml

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update yaml files

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip FP8 test in CLI

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address reviews

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address review comments

Signed-off-by: moraxu <mguzek@nvidia.com>

---------

Signed-off-by: moraxu <mguzek@nvidia.com>

moraxu committed:
[fix] Incorrect mocker argument for a CLI accuracy test in Llama-3.3-70B-Instruct (#4604)

Fix mocker argument

Signed-off-by: moraxu <mguzek@nvidia.com>

Tabrizian committed:
Add missing rcca folder (#4591)

Signed-off-by: Iman Tabrizian <10105175+tabrizian@users.noreply.github.com>

jinyangyuan-nvidia committed:
[fix] Fix Llama4 allgather error due to None tensor (#4511)

* [fix] Fix Llama4 allgather error due to None tensor

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Refactor modifications

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Minor modification

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

* Minor fix

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

---------

Signed-off-by: Jinyang Yuan <154768711+jinyangyuan-nvidia@users.noreply.github.com>

moraxu committed:
[TRTLLM-4932] Add QA accuracy tests for NIM-prioritized models (#4242)

* Add tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add tests v2

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add fixes

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip fp8 test for Ultra

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add tests for Phi

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi - fix

Signed-off-by: moraxu <mguzek@nvidia.com>

* Skip tests for Phi - comment out acc refs

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add more test granularity

Signed-off-by: moraxu <mguzek@nvidia.com>

* Fix examples_test_list.txt

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update test list file

Signed-off-by: moraxu <mguzek@nvidia.com>

* Update yaml files

Signed-off-by: moraxu <mguzek@nvidia.com>

* Address review comments

Signed-off-by: moraxu <mguzek@nvidia.com>

* Remove MMLU tests

Signed-off-by: moraxu <mguzek@nvidia.com>

* Add remaining models

Signed-off-by: moraxu <mguzek@nvidia.com>

---------

Signed-off-by: moraxu <mguzek@nvidia.com>

amirkl94 commented:
/bot run

amirkl94 commented:
@yuxianq Can you verify my `llm_args.py` conflict resolution?

tensorrt-cicd commented:
[PR_Github #6372](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6372/) [ run ] completed with state `ABORTED`

amirkl94 commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6376](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6376/) [ run ] completed with state `ABORTED`

amirkl94 commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6383](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6383/) [ run ] triggered by Bot

yuxianq commented on tensorrt_llm/auto_parallel/auto_parallel.py:
```diff
@@ -149,6 +149,9 @@ def check_dtype(tensor):
 
 
 def auto_parallel(network: Network, config: AutoParallelConfig):
+    logger.warning(
```
Please also cherry-pick the change of the `llm_args.py` file in https://github.com/NVIDIA/TensorRT-LLM/pull/4471 to this PR: https://github.com/NVIDIA/TensorRT-LLM/blob/main/tensorrt_llm/llmapi/llm_args.py#L1362

amirkl94 replied:
Note that the changed lines were removed in https://github.com/NVIDIA/TensorRT-LLM/pull/3823/files#diff-bf140a489175e4c6e5931cd19cc57e06c1634483517f06c4e6fda96d5d26b056L809 , Do you still want me to add the lines from https://github.com/NVIDIA/TensorRT-LLM/pull/4471 ?

yuxianq replied:
`auto_parallel/auto_parallel_world_size` are not removed, they are moved from `LlmArgs` to `TrtLlmArgs`. `LlmArgs` is partitioned into `TorchLlmArgs` and `TrtLlmArgs` now.

tensorrt-cicd commented:
[PR_Github #6383](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6383/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4665](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4665/) completed with status: 'FAILURE'

yuxianq reviewed (approved):
LGTM for the autopp part.

tensorrt-cicd commented:
[PR_Github #6439](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6439/) [ run ] triggered by Bot

MartinMarciniszyn reviewed (approved):
My changes look good.

amirkl94 commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6578](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6578/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #6578](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6578/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4811](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4811/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #6643](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6643/) [ run ] triggered by Bot

tongyuantongyu reviewed (changes requested):
1. Based on https://github.com/NVIDIA/TensorRT-LLM/pull/4455#discussion_r2100513666, 
https://github.com/NVIDIA/TensorRT-LLM/blob/1118dea8401d8224fea2d9c5a031df3b01f819f7/docker/Dockerfile.multi#L180-L182
should be update to
https://github.com/NVIDIA/TensorRT-LLM/blob/d4cccdc48b6a2c23850b9ea8237689e736f3700c/docker/Dockerfile.multi#L175-L181



2. It seems #4545 is not included in this PR. Is it expected?

Naveassaf commented:
/bot skip --comment "Tests passed."

tensorrt-cicd commented:
[PR_Github #6643](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6643/) [ run ] completed with state `ABORTED`
